### ISSUE TITLE
it:test and test:test output files clash

### ISSUE THREAD

**Issue by ebruchez** [scala-js/scala-js#3090]
I am trying to get `it:test` to run. I read https://www.scala-js.org/doc/project/testing.html.

When I run `test:test`, a file named `my-project-test-fastopt.js` is created. So far so good. But when I run `it:test`, the compilation output also goes to `my-project-test-fastopt.js`, and the result is different (as it contains the compilation of different test classes). This does not seem right.

In [`ScalaJSPluginInternal.scala`](https://github.com/scala-js/scala-js/blob/4093c9bf41607320eae260f9d7dbd3c8cf02fc0d/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala#L493), I notice that `-test` is appended. So I tried something like:

```scala
.jsSettings(inConfig(IntegrationTest)(ScalaJSPluginInternal.scalaJSTestSettings ++ (Seq(fastOptJS, fullOptJS) map { packageJSTask =>
    moduleName in packageJSTask := moduleName.value + "-it"
  })): _*)
```

This does create `my-project-it-fastopt.js`, although I still get a `my-project-test-jsdeps.js` (which might be right or not, I don't know, and might not be very relevant nowadays anyway).

Using Scala.js 0.6.19.

**Comment by sjrd** [comment]
What setup do you have to configure `it` with Scala.js to begin with? By default, sbt-scalajs does not configure the `it` configuration at all.

**Comment by ebruchez** [comment]
The setup documented at https://www.scala-js.org/doc/project/testing.html.

**Comment by sjrd** [comment]
Your "workaround" snippet is the right thing to do to give a custom name to the output of

**Comment by ebruchez** [comment]
Thanks. I have been struggling with `it:test` and Scala.js and for now I have reverted to using `test:test` in a separate sbt subproject as I had to get the test working ;) I hope to get back to `it:test` once the dust has settled.

**Comment by sjrd** [comment]
Fixed in 111e414557eb8fb041fc5578531b275b0cb9a7cc.
